app/leaderboard/page.tsx

```diff
@@ -471,50 +543,44 @@ export default function Page() {
                 />
               </div>
             )}
+            {duration !== TOP_50_TAB_STRING &&
+              (address || (!isNoSearchResults && currentSearchedAddress)) &&
+              !isNoSearchResults && (
+                <Button
+                  onClick={() => {
+                    if (checkIfLastPage && !viewMore) {
```

Comment:
can you put this in a onClick handler function? we avoid long inline functions in the codebase